src/Datadog.Trace/Scope.cs

```diff
@@ -39,12 +39,12 @@ internal Scope(Scope parent, Span span, IScopeManager scopeManager, bool finishO
         /// </summary>
         public void Close()
         {
-            _scopeManager.Close(this);
-
             if (_finishOnClose)
             {
                 Span.Finish();
             }
+
+            _scopeManager.Close(this);
```

Comment:
Why the move?

Reply:
Umm. I don't quite remember the motivation for this and I don't see any reason for this now. I'll go ahead and revert this.